app/controllers/users/two_factor_authentication_controller.rb

```diff
@@ -140,9 +140,25 @@ def send_user_otp(method)
 
       job = "#{method.capitalize}OtpSenderJob".constantize
       job_priority = confirmation_context? ? :perform_now : :perform_later
-      job.send(job_priority, code: current_user.direct_otp, phone: phone_to_deliver_to,
-                             otp_created_at: current_user.direct_otp_sent_at.to_s,
-                             locale: user_locale)
+      job.send(job_priority, job_params)
+    end
+
+    def job_params
+      {
+        code: current_user.direct_otp,
+        phone: phone_to_deliver_to,
+        otp_created_at: current_user.direct_otp_sent_at.to_s,
+        message: sms_message,
+        locale: user_locale,
+      }
+    end
+
+    def sms_message
+      if current_user.encrypted_otp_secret_key.present?
+        'jobs.sms_otp_sender_job.login_message'
+      else
+        'jobs.sms_otp_sender_job.create_account_message'
+      end
```

Comment:
Why not put this logic in `SmsOtpSenderJob`? It doesn't belong in the controller. Also, keep in mind that we won't be using jobs anymore, which is what I'm working on in LG-452.

Reply:
Also, that's the wrong logic. `encrypted_otp_secret_key` is for TOTP users. Yes, I know the name of the field is misleading. I would use `SmsLoginOptionPolicy.new(user).configured?` to determine whether or not the user already has an SMS-enabled phone. That way, when we introduce multiple phone numbers, we won't have to change the logic here since the logic should only live in one place: the Policy Object.

Reply:
Ah nice on the policy object!

Reply:
wrt login in SmsOtpSenderJob there is no user in that job.  Would you prefer I passed the user in as a parameter instead?

Reply:
Ah, right. Good point. Maybe we should wait until I refactor this as part of LG-452?

Reply:
Yeah, that's the plan.